rfcs/003-randomness.md

```diff
@@ -0,0 +1,210 @@
+the Local-VRF approach, providing a somewhat trusted and unpredictable
+source of fresh randomness.
+
+There is an implict assumption that the Storage Provider only observes the
+finalized chain
+
+## Introduction
+
+The Polka Storage project requires randomness as a form of creating challenges
+for Storage Providers, with the aim of stopping them from discarding the files
+they are supposed to store, by pre-computing proofs for upcoming challenges.
+Achieving this randomness is not trivial — for example, we cannot allow the
+Storage Provider to generate the randomness alone since they may be dishonest
+and generate values that allow them to fake the storage proofs; furthermore by
+their very nature, blockchains must be somewhat deterministic and as such,
+randomness methods achieved on-chain are not suitable for all use cases. For
+example consider an application that requires multiple rounds of randomness,
+to achieve that you would generally reach out to a pseudo random number
+generator (PRNG), however that PRNG requires a seed to get started, and
+on-chain, everyone can see that seed and as such, predict the outcomes; now
+consider that you changed your code to change the seed every time a new block
+is generated, while the seed is no longer predictable, it is still possible
+for users to read it (it is the latest block hash after all) and predict
+future outcomes while a new block is not available — as such, most methods
+discussed in this RFC are not suitable for cases where multiple random values
+are required (such as casinos where multiple rolls require multiple rounds
+over a single seed), however they are suited for cases where a single random
+value is required.
+
+## Terminology
+
+* SP — Storage Provider
+* VRF — Verifiable Randomness Function
+* PoRep — Proof of Replication
+* PoSt — Proof of Spacetime
+* drand — Distributed Randomness Beacon
+* PRNG — Pseudo Random Number Generator
+
+## The Problem
+
+The Polka Storage Parachain SPs generate proofs that they are, in fact, storing
+the files they committed to store. These proofs are PoRep and PoSt and they sit
+at the core of the system — breaking them means breaking the system, allowing
+storage to be proven without any actual storage being done.
+
+Randomness plays a different role in each proof; for PoRep, the randomness plays
+a crucial role in the generation of a unique sector, avoiding the possibility
+that the storage is outsourced or duplicated to another provider — that is, two
+providers may store the same file, while outsorcing the actual storage to a
+co-located server, effectively scamming the final customer which thinks their
+data is replicated among two distinct providers but in reality, it was only
+stored once.
+
+As for PoSt, randomness is used for challenges, these stop the SP from
+pre-computing proofs an thus, faking their stored files. As long as the SP is
+unable from accurately guessing the next set of challenges, the PoSt should
```

Comment:
unable from where? I assume this must be `As long as the SP is unable to get the randomness accurately guess the next set of challenges`